### PR TITLE
Add missing error messages in store_credit model

### PR DESCRIPTION
**Pull request by nishant-cyro**
Error messages are not present in en.yml for store credit model.

Done with little bit refactoring.

### CODE REVIEW

File: core/config/locales/en.yml [hunk 1]
```diff
@@ -323,14 +323,25 @@ en:
...
+          attributes:
+            amount_used:
+              cannot_be_greater_than_amount: Cannot be greater than amount.
+              greater_than_zero_restrict_delete: is greater than zero. Can not delete store credit.
+            amount_authorized:
+              exceeds_total_credits: Exceeds total credits.
+        spree/variant:
           attributes:
             base:
-              required_shipping_category: "You must select at least one shipping category"
```
Comment (damianlegawiec) [inline core/config/locales/en.yml]: @nishant-cyro why did you remove this one?

Comment (nishant-cyro) [inline core/config/locales/en.yml]: @damianlegawiec - I haven't removed this one, just moved it above spree/store.
